src/de/roderick/weberknecht/WebSocketMessage.java

```diff
@@ -18,12 +18,16 @@
 
 public class WebSocketMessage {
   private Byte[] message;
+  private boolean asText;
 
-  public WebSocketMessage(final Byte[] message) {
+  public WebSocketMessage(final Byte[] message, boolean asText) {

```

Comment:
Here and for the field, perhaps something like `uses text format`?

Or, we could consider changing `WebSocketMessage` to be abstract, and having two conrete implementations, one for text and one for binary formats.